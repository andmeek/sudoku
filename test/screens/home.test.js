import { shallowMount } from '@vue/test-utils'
import Home from '../../src/screens/home.vue'

describe('Home.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(Home)
  })

  test('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('emits an "newgame" event when the user clicks new game', () => {
    wrapper.find({ ref: 'newgame' }).trigger('click')

    expect(wrapper.emitted()['new-game']).toBeTruthy()
  })

  test('emits an "stats" event when the user clicks stats', () => {
    wrapper.setData({ showStats: true })
    wrapper.find({ ref: 'stats' }).trigger('click')

    expect(wrapper.emitted().stats).toBeTruthy()
  })

  test('enables the stats button when showStats is enabled', async () => {
    wrapper.setData({ showStats: true })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
