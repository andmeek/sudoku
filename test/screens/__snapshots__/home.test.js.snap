// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue enables the stats button when showStats is enabled 1`] = `
<div class="home screen">
  <h1>Play Sudoku</h1>
  <div class="actions"><button class="menu">
      New Game
    </button> <button class="menu">
      Stats
    </button> <a href="https://sudoku.com/how-to-play/sudoku-rules-for-complete-beginners/" target="_blank" class="menu">How To Play</a> <a href="https://github.com/andmeek/sudoku" target="_blank">Source Code on Github</a></div>
</div>
`;

exports[`Home.vue matches snapshot 1`] = `
<div class="home screen">
  <h1>Play Sudoku</h1>
  <div class="actions"><button class="menu">
      New Game
    </button> <button disabled="disabled" class="menu">
      Stats
    </button> <a href="https://sudoku.com/how-to-play/sudoku-rules-for-complete-beginners/" target="_blank" class="menu">How To Play</a> <a href="https://github.com/andmeek/sudoku" target="_blank">Source Code on Github</a></div>
</div>
`;
